### PR TITLE
Okta integration

### PR DESCRIPTION
**Pull request by rfox2**
This is deceptively simple I think. Okta integration at the basic level can be accomplished by a 1 to 1 swap of Devise CAS with omniauth / okta. This set of changes does not take advantage of the Open ID Connect attributes that are passed back to the callback method - that would be the next logical step to see if we can completely cut the dependency on LDAP.

### CODE REVIEW

File: app/views/layouts/_user_links.html.erb [hunk 1]
```diff
@@ -13,7 +13,13 @@
...
     <ul class="nav nav-pills">
       <li class="dropdown">
-        <%= link_to("Log In", user_session_path())%>
+
+        <%= link_to("Log In", user_oktaoauth_omniauth_authorize_path())%> 
+        <%
+=begin%>
+ <%= link_to("Log In", user_session_path())%> 
+<%
+=end%>
```
Comment (mlkirkpatrick) [inline app/views/layouts/_user_links.html.erb]: Do we really need this hanging around in the source?

File: config/application.rb [hunk 2]
```diff
@@ -81,7 +81,7 @@ class Application < Rails::Application
         :deliver_with => :deliver_now,
         :email_prefix => "[Factotum #{Rails.env}] ",
         :sender_address => %{"Factotum Notifier" <factotum@nd.edu>},
-        :exception_recipients => %w{jkennel@nd.edu jhartzler@nd.edu rfox2@nd.edu dwolfe2@nd.edu}
+        :exception_recipients => %w{rfox2@nd.edu}
```
Comment (mlkirkpatrick) [inline config/application.rb]: Did you mean to change this, or was this just for testing?

File: config/environments/development.rb [hunk 3]
```diff
@@ -50,7 +50,7 @@
   # Custom configuration
   config.base_url = "http://localhost:3003"
   config.library_url = "http://librarypprd.nd.edu"
-  config.cas_base = 'https://login-test.cc.nd.edu/cas'
+  # config.cas_base = 'https://login-test.cc.nd.edu/cas'
```
Comment (mlkirkpatrick) [inline config/environments/development.rb]: Again, I'm generally against dead code being left in place commented out. If there is no intention to enable it in the future, it just bloats the file. If it's there for historical reasons, that's what git history is for.

File: config/application.rb (hunk 2 above)
Comment (rfox2) [reply config/application.rb]: Meant to change it

**Pull request merged**